lstm.go

```diff
@@ -1,40 +1,132 @@
 package golgi
 
 import (
+	"fmt"
+
 	"github.com/chewxy/hm"
-	G "gorgonia.org/gorgonia"
+	"gorgonia.org/gorgonia"
 	"gorgonia.org/tensor"
 )
 
-// LSTM represents a static LSTM model
-type LSTM struct{}
+func newLSTM(g *gorgonia.ExprGraph, layer *LSTMLayer, name string) (lp *LSTM) {
+	var l LSTM
+	l.g = g
+	l.name = name
+	l.input = layer.getInput(g, name)
+	l.forget = layer.getInput(g, name)
+	l.output = layer.getInput(g, name)
+	l.cell = layer.getInput(g, name)
+	return &l
+}
+
+// LSTM represents an LSTM RNN
+type LSTM struct {
+	name string
+
+	g *gorgonia.ExprGraph
+
+	input  whb
+	forget whb
+	output whb
+	cell   whb
+}
+
+func (l *LSTM) getGate(w *whb, inputVector, prevHidden *gorgonia.Node, act ActivationFunction) (gate *gorgonia.Node, err error) {
+	var h0 *gorgonia.Node
+	if h0, err = gorgonia.Mul(w.wx, inputVector); err != nil {
+		return
+	}
+
+	var h1 *gorgonia.Node
+	if h1, err = gorgonia.Mul(w.wh, prevHidden); err != nil {
+		return
+	}
+
+	// Set gate as the sum of h0 and h1
+	if gate, err = gorgonia.Add(h0, h1); err != nil {
+		return
+	}
+
+	// Set the gate as the sum of current gate and the whb bias
+	if gate, err = gorgonia.Add(gate, w.b); err != nil {
+		return
+	}
+
+	// Return gate with activation func performed on it
+	return act(gate)
+}
 
 // Model will return the gorgonia.Nodes associated with this LSTM
-func (l *LSTM) Model() G.Nodes {
-	panic("not implemented")
+func (l *LSTM) Model() gorgonia.Nodes {
+	return gorgonia.Nodes{l.input.b, l.forget.b, l.output.b, l.cell.b}
 }
 
 // Fwd runs the equation forwards
-func (l *LSTM) Fwd(x G.Input) G.Result {
-	panic("not implemented")
+// TODO: Convert this to a proper Fwd, this is still a crude copy of charRNN
+func (l *LSTM) Fwd(x gorgonia.Input) gorgonia.Result {
+	var (
+		inputGate *gorgonia.Node
+		err       error
+
+		// TODO: These need to be set
```

Comment:
Indeed - these "dummy" vectors can be set as part of the layer's unexported fields.